src/nwb_datajoint/decoding/clusterless.py

```diff
@@ -63,84 +83,72 @@ class UnitMarks(dj.Computed):
+
+        WAVEFORM_PARAMS = {
+            'ms_before': 0.5,
+            'ms_after': 0.5,
+            'max_spikes_per_unit': None,
+            'n_jobs': 5,
+            'total_memory': '5G',
+        }
+        waveform_extractor = si.extract_waveforms(
+            recording=recording,
+            sorting=sorting,
+            folder=waveform_extractor_path,
+            **WAVEFORM_PARAMS)
 
         if mark_param['mark_type'] == 'amplitude':
-            # get the marks and timestamps
-            #marks = np.empty((0, 4), dtype='int16')
-            # more flexible if there are dead channels
-            marks = np.empty((0, waveforms[0][0].shape[0]), dtype='int16')
-            timestamps = np.empty((0), dtype='float64')
-            for index, unit in enumerate(waveforms):
-                marks = np.concatenate(
-                    (marks, np.amin(np.asarray(unit, dtype='int16'), axis=2)), axis=0)
-                timestamps = np.concatenate(
-                    (timestamps, nwb_units.loc[units['unit_id'][index]].spike_times), axis=0)
-            # sort the timestamps to order them properly
-            sort_order = np.argsort(timestamps)
-            timestamps = timestamps[sort_order]
-            marks = marks[sort_order, :]
-
-            if 'threshold' in mark_param_dict:
-                print('thresholding')
-                # filter the marks by the amplitude threshold
-                if mark_param_dict['sign'] == -1:
-                    include = np.where(
-                        np.amax(marks, axis=1) <= (mark_param_dict['sign'] *
-                                                   mark_param_dict['threshold'])
-                    )[0]
-                elif mark_param_dict['sign'] == -1:
-                    include = np.where(np.amax(marks, axis=1)
-                                       >= mark_param_dict['threshold'])[0]
-                else:
-                    include = np.where(
-                        np.abs(np.amax(marks, axis=1)) >= mark_param_dict['threshold'])[0]
-                timestamps = timestamps[include]
-                marks = marks[include, :]
-
-            # create a new AnalysisNwbfile and a timeseries for the marks and save
-            key['analysis_file_name'] = AnalysisNwbfile().create(
-                key['nwb_file_name'])
-            nwb_object = pynwb.TimeSeries(
-                name='marks',
-                data=marks,
-                unit='uV',
-                timestamps=timestamps,
-                description=f'amplitudes of spikes from electrodes {recording.get_channel_ids()}')
-            key['marks_object_id'] = (AnalysisNwbfile().add_nwb_object(
-                key['analysis_file_name'], nwb_object))
-            AnalysisNwbfile().add(key['nwb_file_name'],
-                                  key['analysis_file_name'])
-            self.insert1(key)
+            marks = np.concatenate(
+                [UnitMarks._get_peak_amplitude(
```

Comment:
That's a good point. I wasn't aware of this function. Let me compare the output of this function and see how hard it will be to get in there. Again, if it is more work I would prefer to put this in a separate PR since this will not involve table dropping.

Reply:
I added some more robust code for estimating the peak amplitude in cases where the peak amplitude isn't the center value of the waveform.